### PR TITLE
Use .env to specify port and base URL

### PR DESCRIPTION
**Pull request by ryderwishart**
My port 5000 was occupied when I tried to run this, and I found the port was hard coded in several places. 

This tiny PR uses a `metaconfig.json` file where the user can specify the port and the base URL if needed.

### CODE REVIEW

**Comment by ryderwishart** [comment]
Sure, the environment option seems good. I think it should probably increment in case an exception is raised as well when there is no port specified in the `.env`.

**Comment by joshsny** [comment]
Great! I have made a start on the `.env` implementation so pushed changes here, but agree with you that incrementing if there is no specified port would be best.

**Comment by ryderwishart** [comment]
@joshsny to test locally, use `nc -l 5000` or whatever to listen on the default port, and then reset db and run world. 

Apple airplay uses 5000 in some strange way, so I set the default to 8080 and tested by listening on 8080 and 8081. 

To check if the ports are truly in use, list open files on the ports you are listening on, e.g.: `lsof -i :5000`.

Let me know what you think about this approach, and I can add the base URL as another env variable if you like this method.

File: src/utils/general.py [hunk 1]
```diff
@@ -19,6 +20,14 @@ def deduplicate_list(items: list[T], key: Callable[[T], K]) -> list[T]:
...
     return str(UUID(hashlib.sha1(seed.encode()).hexdigest()[:32]))
 
+
 def is_port_in_use(port):
     with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
-        return s.connect_ex(('localhost', port)) == 0
+        return s.connect_ex(("localhost", port)) == 0
+
+
+def get_open_port():
```
[the remark this answers is not in the document]
Comment (joshsny) [reply src/utils/general.py:29]: Looks good!

File: src/main.py [hunk 2]
```diff
@@ -56,7 +57,7 @@ def run_world():
 
 def run_server():
     app = get_server()
-    run_in_new_loop(app.run_task())
+    run_in_new_loop(app.run_task(port=int(os.getenv("PORT", "5000"))))
```
Comment (joshsny) [inline src/main.py:60]: This needs to use the get_open_port function

File: src/utils/windowai_model.py [hunk 3]
```diff
@@ -86,7 +87,8 @@ def _call(
             )
 
         ws = websocket.WebSocket()
-        ws.connect("ws://127.0.0.1:5000/windowmodel")
+        port = get_open_port()
```
Comment (joshsny) [inline src/utils/windowai_model.py:90]: @ryderwishart Won't the current port be closed, causing this port to be different to the one in `main.py`?

**Review by joshsny — CHANGES_REQUESTED** [review]
Couple minor comments but implementation looks great

File: src/utils/windowai_model.py (hunk 3 above)
Comment (ryderwishart) [reply src/utils/windowai_model.py:90]: Good point. Perhaps we can set something like `ACTIVE_PORT` dynamically once an open port is found, and then check for `ACTIVE_PORT` initially on any other calls.

Comment (ryderwishart) [reply src/utils/windowai_model.py:90]: @joshsny see new commits below.

**Review by joshsny — APPROVED** [review]

**Pull request merged**